src/Command/Ssh/SshKeyCommandBase.php

```diff
@@ -151,33 +148,99 @@ protected function pollAcquiaCloudUntilSshSuccess(
   ): void {
     // Create a loop to periodically poll the Cloud Platform.
     $loop = Loop::get();
-    $spinner = LoopHelper::addSpinnerToLoop($loop, 'Waiting for the key to become available on the Cloud Platform', $output);
-
-    // Wait for SSH key to be available on a web.
+    $spinner_git = LoopHelper::addSpinnerToLoop($loop, 'Waiting for the key to become available in Cloud Platform git', $output);
+    $spinner_nonprod = LoopHelper::addSpinnerToLoop($loop, 'Waiting for the key to become available in Cloud Platform non-prod environments', $output);
+    $spinner_prod = LoopHelper::addSpinnerToLoop($loop, 'Waiting for the key to become available in Cloud Platform prod environments', $output);
     $cloud_app_uuid = $this->determineCloudApplication(TRUE);
-    $environment = $this->getAnyNonProdAhEnvironment($cloud_app_uuid);
+    $permissions = $this->cloudApiClientService->getClient()->request('get', "/applications/{$cloud_app_uuid}/permissions");
+    $perms = array_column($permissions, 'name');
+    $vcs_url = $environment_nonprod = $environment_prod = NULL;
+    if (in_array('add ssh key to git', $perms, TRUE)) {
+      $poll_git = TRUE;
+      $full_url = $this->getAnyVcsUrl($cloud_app_uuid);
+      $url_parts = explode(':', $full_url);
+      $vcs_url = $url_parts[0];
+    }
+    else {
+      $poll_git = FALSE;
+      $output->writeln('<comment>You do not have access to Cloud Platform git on this application and will not be able to clone your codebase to this IDE. Check that you have the <options=bold>add ssh key to git</> permission. Documentation on Cloud Teams permissions: <href=https://docs.acquia.com/cloud-platform/access/teams/permissions/default/>https://docs.acquia.com/cloud-platform/access/teams/permissions/default/</>');
```

Comment:
Let's use `$this->io`

Reply:
`$this->io` seems really broken. Depending on the method (warning vs comment, for instance), it won't respect inline styles (`<options=bold>`). And it keeps breaking the URL into multiple lines for no apparent reason.

Reply:
Yeah. That's the downside of it. It basically takes over all formatting. That's why I've ended up mixing it with `$output->writeln()`.